core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala

```diff
@@ -112,11 +112,12 @@ object SparkHadoopWriter extends Logging {
       jobTrackerId, sparkStageId, sparkPartitionId, sparkAttemptNumber)
     committer.setupTask(taskContext)
 
-    val (outputMetrics, callback) = initHadoopOutputMetrics(context)
-
     // Initiate the writer.
     config.initWriter(taskContext, sparkPartitionId)
     var recordsWritten = 0L
+    
+    // Initialize callback function after the writer.
```

Comment:
The comment is better to explain why we must initialize the callback here. E.g., `We must initialize the callback for calculating bytes written after the statistic table is initialized in FileSystem`.

Reply:
ok